app/controllers/organizations/service_agreements_controller.rb

```diff
@@ -57,6 +56,10 @@ def create
 
     private
 
+    def agreement_creator
+      @agreement_creator ||= ServiceAgreementCreator.new(current_organization_admin, service_agreement_params)
+    end
+
     def service_specifications
       @service_specifications ||= current_organization.service_specifications
     end
```

Comment:
This method is also not really necessary (in that state).
Maybe have a `before_action :load_service_specifications, only: %i[new create]`
where you have
```suggestion
    def load_service_specifications
      @service_specifications = current_organization.service_specifications
    end
```